woocommerce/woocommerce
# Restore stock_status in V3 response.

### All Submissions:






### Changes proposed in this Pull Request:



Field `stock_status` is special in that [it's computed using another field `in_stock`](https://github.com/woocommerce/woocommerce/blob/master/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php#L1380) (present in V2 API). When we added the performance enhancements to fetch only what was requested, this field got skipped because `in_stock` was never going to be asked by V3 schema (since it was specific to V2), and without in_stock, stock_status can't be computed.

Quite similarly, in Orders V3, `_gmt` dates fields depends upon their usual without `_gmt` fields, and cannot be computed unless their original fields are present.

Also returns refund by default even when empty for consistency with earlier versions.  

Additionally, this PR also adds intensive tests for V2 and V3 orders and products endpoint, to verify that all expected fields are returned, and all expected fields return when they are the only field fetched in the response.

Closes #28729.

### How to test the changes in this Pull Request:

1. Fetch product response in API V3
2. Make sure the field `stock_status` is present.
3. Fetch any order using API and with param `_fields=date_created_gmt`. Make sure that this field is returned.

### Other information:




### Changelog entry

> Fix - Restore `stock_status` in V3 response.
